pkg/privilege/privileges/cache.go

```diff
@@ -1145,10 +1166,12 @@ func (record *tablesPrivRecord) match(user, host, db, table string) bool {
 }
 
 func (record *columnsPrivRecord) match(user, host, db, table, col string) bool {
+	// `SELECT COUNT(*) ...` requires a column-level SELECT privilege of any column,
+	// so we add a special case "*" here
 	return record.baseRecord.match(user, host) &&
 		strings.EqualFold(record.DB, db) &&
 		strings.EqualFold(record.TableName, table) &&
-		strings.EqualFold(record.ColumnName, col)
+		(strings.EqualFold(record.ColumnName, col) || col == "*")
```

Comment:
The requirement is to match **any** column.
But here it actually match **the first** column.

For example, we have three columns a, b, c where
a: no privilege
b: write-only privilege
c: read-only privilege

The match function returns record of b (the first match) ... but it's not correct.
Later, columnPriv&select_priv would return 0 and the check fail, right?

Reply:
See https://github.com/pingcap/tidb/pull/61677/files#diff-c3393c2c85014927cf2d80a965fb61f15fdc3f375d9deefeb167ef3057c90f7eR1303